### DISCUSSION TITLE
DgsEntityFetcher optimization

### DISCUSSION THREAD

**Discussion by jacobo-cloudbeds** [Netflix/dgs-framework#1210] [Q&A]
Hi!

I'm testing GraphQL Federation with DGS and while replicating the example from the docs I realized that `movie` and `reviewsFetcher` are called N times (once per Show), but reviews could be fetched for all shows in a single query more efficiently. I guess I could use the `DgsDataFetchingEnvironment` to get access to the `representations` argument the first time `movie` is called, fetch all reviews for all shows, and store them in the context to return them later when `reviewsFetcher` is called. I wonder if there is a better/more elegant way to do it.

```
    @DgsEntityFetcher(name = "Show")
    public Show movie(Map<String, Object> values) {
        return new Show((String) values.get("id"), null);
    }

    @DgsData(parentType = "Show", field = "reviews")
    public List<Review> reviewsFetcher(DgsDataFetchingEnvironment dataFetchingEnvironment)  {
        Show show = dataFetchingEnvironment.getSource();
        return reviews.get(show.getId());
    }
```

Thanks!

**Comment by srinivasankavitha** [comment]
Have you considered using a data loader which will batch all the calls for
fetching reviews. You won't be able to avoid the N calls to reviewsFetcher,
but reviewsFetcher can use a dataloader which will optimize the fetching
from other backends/datastores etc.

**Reply by tknupf** [reply]
Am I correct that this is still not possible?
If not I will DgsDataFetchingEnvironment.getVariables().get("representations") parse the information out of it
and store the other results in the customContext.

I will not use DataLoaders because that breaks the federation.

**Reply by srinivasankavitha** [reply]
You should be able to use a data loader to load all the results for a single field in the entity fetcher. If you need it for multiple different fields, then yes, you'll have to do what you described.

**Reply by tknupf** [reply]
O.k. so how do I return all the values for the above example?
